File: apps/backend/src/services/profile.service.ts
import { prisma } from '../lib/prisma.js';

export interface ProfileData {
  role: 'job_seeker' | 'sales';
  name?: string; // User's name from onboarding
  displayName?: string;
  timeZone?: string;
  reminderMinutes?: number;
  // Job Seeker fields
  resumeUrl?: string;
  jobRole?: string;
  targetCompanies?: string[];
  // Sales fields
  companyName?: string;
  productDescription?: string;
  salesPainPoints?: string;
  salesTargets?: string;
  // Shared preferences
  preferredMeetingTypes?: string[];
  notificationPreference?: 'push' | 'email' | 'both';
  notes?: string;
  // Additional flexible data
  profileData?: Record<string, any>;
}

export class ProfileService {
  /**
   * Create or update user profile
   */
  async upsertProfile(userId: string, data: ProfileData) {
    // Update user's name if provided
    if (data.name) {
      await prisma.user.update({
        where: { id: userId },
        data: { name: data.name },
      });
    }

    const profile = await prisma.profile.upsert({
      where: { userId },
      update: {
        role: data.role,
        displayName: data.displayName,
        timeZone: data.timeZone,
        reminderMinutes: data.reminderMinutes ?? 30,
        resumeUrl: data.resumeUrl,
        jobRole: data.jobRole,
        targetCompanies: data.targetCompanies,
        companyName: data.companyName,
        productDescription: data.productDescription,
        salesPainPoints: data.salesPainPoints,
        salesTargets: data.salesTargets,
        preferredMeetingTypes: data.preferredMeetingTypes,
        notificationPreference: data.notificationPreference ?? 'push',
        notes: data.notes,
        profileData: data.profileData,
      },
      create: {
        userId,
        role: data.role,
        displayName: data.displayName,
        timeZone: data.timeZone,
        reminderMinutes: data.reminderMinutes ?? 30,
        resumeUrl: data.resumeUrl,
        jobRole: data.jobRole,
        targetCompanies: data.targetCompanies ?? [],
        companyName: data.companyName,
        productDescription: data.productDescription,
        salesPainPoints: data.salesPainPoints,
        salesTargets: data.salesTargets,
        preferredMeetingTypes: data.preferredMeetingTypes ?? [],
        notificationPreference: data.notificationPreference ?? 'push',
        notes: data.notes,
        profileData: data.profileData,
      },
      include: {
        user: {
          select: {
            id: true,
            email: true,
            name: true,
          },
        },
      },
    });

    return profile;
  }

  /**
   * Get user profile
   */
  async getProfile(userId: string) {
    const profile = await prisma.profile.findUnique({
      where: { userId },
      include: {
        user: {
          select: {
            id: true,
            email: true,
            name: true,
            createdAt: true,
          },
        },
      },
    });

    return profile;
  }

  /**
   * Delete user profile
   */
  async deleteProfile(userId: string) {
    await prisma.profile.delete({
      where: { userId },
    });
  }
}

export const profileService = new ProfileService();

File: apps/ios/Readi/Services/ProfileService.swift
//
//  ProfileService.swift
//  Readi
//
//  Created by Readi Team
//

import Foundation

struct ProfileRequest: Codable {
    let name: String? // User's name from onboarding
    let role: String
    let displayName: String?
    let timeZone: String?
    let reminderMinutes: Int?
    let resumeUrl: String?
    let jobRole: String?
    let targetCompanies: [String]?
    let companyName: String?
    let productDescription: String?
    let salesPainPoints: String?
    let salesTargets: String?
    let preferredMeetingTypes: [String]?
    let notificationPreference: String?
    let notes: String?
    
    // No CodingKeys - using default camelCase to match backend
}

struct ProfileResponse: Codable {
    let profile: UserProfile
}

class ProfileService {
    static let shared = ProfileService()
    private init() {}
    
    private let apiClient = APIClient.shared
    
    func createProfile(_ request: ProfileRequest) async throws -> UserProfile {
        let response: ProfileResponse = try await apiClient.post("/profiles", body: request)
        return response.profile
    }
    
    func getProfile() async throws -> UserProfile {
        let response: ProfileResponse = try await apiClient.get("/profiles/me")
        return response.profile
    }
    
    func updateProfile(_ request: ProfileRequest) async throws -> UserProfile {
        let response: ProfileResponse = try await apiClient.put("/profiles", body: request)
        return response.profile
    }
}

